### ISSUE TITLE
[BUG] - Shopping list items of the same type are not consolidated

### ISSUE THREAD

**Issue by Yuni1988** [mealie-recipes/mealie#3417]
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [X] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

I created a meal plan with breakfast, lunch and dinner recipes selected. When I add the items to the shopping list via the meal planner feature, the items on the shoppinglist are not consolidated. For example, I have several entries for eggs and several entries for salt. 

### Steps to Reproduce

I'm attaching images that show the workflow.
1.) Fill a day in mealplanner with recipes
2.) Create a shopping list or empty an existing one
3.) Add items from the mealplanner to the shopping list
![CreateShoppingList](https://github.com/mealie-recipes/mealie/assets/69676522/c19cbe34-b658-40a9-958c-82b3f7794caa)
![IngredientList](https://github.com/mealie-recipes/mealie/assets/69676522/96ef1b2c-8a5f-4276-9610-c0665a53afdc)
![ItemsBefore](https://github.com/mealie-recipes/mealie/assets/69676522/8f89f4a1-8be5-4435-8521-4ac95ec0512e)

If I them select and de-select items on the shopping list, they will get summed together. So it looks like this step is skipped when adding items via the mealplanner.
![ItemsAfter](https://github.com/mealie-recipes/mealie/assets/69676522/8687b896-494f-4c4e-8d69-f1fd4e6e10f5)

### Please provide relevant logs

The log is empty when I perform these actions.

### Mealie Version

v1.4.0, f709d11952ca8552214e4b9e8c2a189bc11bfb2a 
Also reproducible on the demo platform.

### Deployment

Docker (Linux)

**Comment by michael-genson** [comment]
Looks like this is specific to adding from the mealplanner; adding a single recipe is fine, even when it has multiple repeated ingredients (e.g. 1 clove garlic, 2 cloves garlic).

My guess is that it's a race condition due to sending multiple "add recipe" requests at once, and they don't see each other. Solution would be to either send one request at a time (slow) or add a new backend API to accept multiple recipes in one request

**Comment by boc-the-git** [comment]
Marking confirmed based on Michael's comment

**Comment by padawarmik** [comment]
Hi, 
I have the same problem with version 2.1.0. 
Is there any update regarding this bug?

**Comment by boc-the-git** [comment]
@padawarmik are you able to share your exact replication steps?
This is definitely an edge case bug, so I'm keen to have as much documentation about the scenarios as possible.

**Comment by padawarmik** [comment]
@boc-the-git 
Sure, to reproduce you can make a plan for a day in a meal planner and then try to add an entire day to the shopping list. 

![image](https://github.com/user-attachments/assets/41c7c7d6-e46c-4f7b-94bc-a4639ea06e44)

and here is a part of this shopping list:

![image](https://github.com/user-attachments/assets/3b20d75a-3d30-40f7-8b45-df5ff8537c8b)

It happens every time I do the shopping list from the meal planner. I was unable to proceed without replicating the issue.  
If I can do something more, let me know. I would be glad to help.

**Comment by destratified** [comment]
agreed.  this is still a reproducible bug following the above steps

**Comment by michael-genson** [comment]
I can't reproduce this. What version are you on?

**Comment by destratified** [comment]
Build
[3a237258a11349987bf9c790cd696aaec332432d]
Nightly

ill update you when ive done the following meal plan > list execution as the build # has changed since the report

**Comment by michael-genson** [comment]
Yeah if you could provide some logs or a video capture or something that would be great. Bonus points if you can reproduce on the demo.
